### PR TITLE
CTIA: independant ES target properties

### PR DESCRIPTION
**Pull request by ereteog**
> Related #threatgrid/iroh#3072

This PR introduces the ability to migrate stores across different clusters and more flexibility for the target store configuration:
- The base change is to enable configuring dedicated properties for target stores instead of building them from source stores.
- It also get rid of the full source initialization and use only necessary pure init function that do not update templates and mappings.
- It put back the command arguments for `confirm` and `restart` params which is more handy to use for OPS. In particular it enable OPS to restart a migration with the same properties file.
- Updates the doc with the full new process.

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

### CODE REVIEW

File: doc/migration.md [hunk 1]
```diff
@@ -1,18 +1,140 @@
- There is a dedicated task to migrate data from prior versions of CTIA. 
- This task will run through all configured stores, transform and copy data to new elasticsearch indices. 
- A migration state will be stored in a configured state to enable restart.
+There is a dedicated task to migrate data from prior versions of CTIA. 
+This task will run through all configured stores, transform and copy data to new Elasticsearch indices. This is possible to migrate these data in another cluster.
```
Comment (frenchy64) [inline doc/migration.md:2]: Last sentence could be clearer. Perhaps `It is possible to migrate data between clusters.`.

File: doc/migration.md [hunk 2]
```diff
@@ -1,18 +1,140 @@
...
+ctia.migration.store.es.default.refresh_interval=1s
+ctia.migration.store.es.default.default_operator=AND
+```
+Then you can specify Elasticsearch properties for each target store.
+```
+ctia.migration.store.es.sighting.indexname=v1.2.0_ctia_sighting
+ctia.migration.store.es.sighting.rollover.max_docs=100000000
+```
+
+When a parameter is not specified for target store neither in it's ES configuration nor in the default migration ES configuration, the source parameters are reused.
```
Comment (frenchy64) [inline doc/migration.md:49]: it's => its

File: test/ctia/task/migration/migrate_es_stores_test.clj [hunk 3]
```diff
@@ -654,11 +656,15 @@
...
-          (is (= expected-indices
-                 (select-keys formatted-cat-indices
-                              (keys expected-indices))))
-
+                                                                (:port default))
+              result-indices (select-keys formatted-cat-indices
+                                          (keys expected-indices))]
+          (is (= expected-indices result-indices)
+              (let [[only-expected only-result _]
+                    (clojure.data/diff expected-indices result-indices)]
```
Comment (frenchy64) [inline test/ctia/task/migration/migrate_es_stores_test.clj:664]: Please add `clojure.data` require.

**Review by frenchy64 — APPROVED** [review]
Lots of nice changes, thanks!

File: test/ctia/properties_test.clj [hunk 4]
```diff
@@ -0,0 +1,38 @@
+(ns ctia.properties-test
+  (:require [ctia.properties :as sut]
+            [clojure.test :refer [deftest is]]
+            [clj-momo.lib.es.schemas :refer [Refresh]]
+            [schema.core :as s]))
+
+(deftest es-store-impl-properties-test
+  (is (= {"ctia.store.es.malware.host" s/Str
```
Comment (vemv) [inline test/ctia/properties_test.clj:8]: Can you add some `testing`s?

File: test/ctia/task/migration/store_test.clj [hunk 5]
```diff
@@ -875,3 +889,85 @@
...
+                            :refresh_interval "10s"}
+        base-es-props {:ctia {:store {:es {:default default-es-props}}}}
+        with-source-props (assoc-in base-es-props
+                                    [:ctia :store :es :sighting]
+                                    source-custom-props)
+        with-migration-cluster-props (assoc-in with-source-props
+                                               [:ctia :migration :store :es :default]
+                                               migration-cluster-props)]
+
+    (testing "simple migration properties that preserves ES source properties and simply adds a prefix to newly created index"
```
Comment (vemv) [inline test/ctia/task/migration/store_test.clj:925]: Ideally `testing`s are complete sentences, e.g. subject + predicate. Without a predicate they feel incomplete, as the behavior being tested is not described.

**Pull request merged**